### PR TITLE
feat: refactor cucumber tests and updates tari dependency 

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description
---
1. Corrects a failing cucumber step
2. Updates the Tari version to pre-release 2
3. Moves cucumber tests to its own crate.

Motivation and Context
---

How Has This Been Tested?
---

What process can a PR reviewer use to test or verify this change?
---

Breaking Changes
---

- [x] None
- [ ] Requires data directory to be deleted
- [ ] Other - Please specify

### CODE REVIEW

**Comment by sdbondi** [comment]
We should actually be using `v0.51.0-pre.0`

File: applications/tari_dan_app_utilities/Cargo.toml [hunk 1]
```diff
@@ -6,10 +6,10 @@ edition = "2021"
 license = "BSD-3-Clause"
 
 [dependencies]
-tari_common_types = { git = "https://github.com/tari-project/tari.git", tag = "v0.50.0-pre.1" }
-tari_core = { git = "https://github.com/tari-project/tari.git", tag = "v0.50.0-pre.1", default-features = false, features = ["transactions"] }
+tari_common_types = { git = "https://github.com/tari-project/tari.git", tag = "v0.50.0-pre.2" }
```
Comment (sdbondi) [inline applications/tari_dan_app_utilities/Cargo.toml]: We should be using the latest tag `v0.51.0-pre.0`

Comment (jorgeantonio21) [reply applications/tari_dan_app_utilities/Cargo.toml]: Thanks, we will have to update to a newer tag, though

**Review by sdbondi — DISMISSED** [review]
ACK

Looks like mining has slowed down a lot in cucumbers with the base layer upgrade. I did look into it a little and it seemed to be the wallet call to create a coinbase is very slow.

**Review by sdbondi — DISMISSED** [review]

**Review by sdbondi — APPROVED** [review]

**Pull request merged**